src/Components/App/App.js

```diff
@@ -14,34 +14,56 @@ const App = () => {
   const [favorites, setFavorites] = useState([])
   const [loading, setLoading] = useState(true)
   const [error, setError] = useState(null)
-
-const thisData = mockData.data.map((data) => {
-  
-})
+  const [truckLocation, setTruckLocation] = useState([])
+  const [userLocation, setUserLocation] = useState('')
 
   const favTruck = (truck) => {
     if (!favorites.find((fav) => fav.id === truck.id)) {
       const newFavState = [...favorites, truck]
-      setFavorites(newFavState)
+        setFavorites(newFavState)
     } else {
       const newFavState = favorites.filter(fav => {
         return fav.id !== truck.id
       })
-      setFavorites(newFavState)
+        setFavorites(newFavState)
     }
   }
 
-  const removeFav = (truck) => {
+const importDistance = (miles) => {
+  setTruckLocation(...miles)
+  console.log(truckLocation, 'truckLocation')
+  return miles
+}
+
+  const removeFav =  (truck) => {
 // filter over vendors and !== the paramter truck and reset state
 const newFavState = favorites.filter((fav) => fav.id !== truck.id)
-setFavorites(newFavState)
+  setFavorites(newFavState)
   }
 
   const searchResults = (searchValue) => {
+    let lowerSearchValue = searchValue.toLowerCase()
+    let nameSearchResults = vendors.filter((v) => v.name.toLowerCase().includes(lowerSearchValue) || v.tags.toLowerCase().includes(lowerSearchValue) || v.description.toLowerCase().includes(lowerSearchValue))
+    setVendors(nameSearchResults)
     //this is going to take in whatever the search value and based upon the search value it will iterate over the vendors and update the vendors state accordingly
 
     // now if we are passing down the name of a  truck or location of truck then we just need to iteratre over vendors and filter where we see fit
   }
+const searchButtons = (event, button) => {
+  console.log(button, 'in app')
+  event.preventDefault();
+  if (button === "favorites") {  
+    setVendors(favorites)
+    // setVendors(vendors.filter(v => v.favorited === true))
+  } 
+  if (button === "openNow") {
+    setVendors(vendors.filter(v => v.status === "true"))
+  }
+  if (button === "closest") {
```

Comment:
I'll keep thinking about this and see if I come up with a good solution for sorting, here!